synapseclient/api/entity_services.py

```diff
@@ -6,8 +6,6 @@
 from dataclasses import dataclass
 from typing import TYPE_CHECKING, Any, AsyncGenerator, Dict, List, Optional, Union
 
-from async_lru import alru_cache
-
```

Comment:
I decided to drop the `alru_cache` here. I saw some errors when I was testing and removing this gave me a bit more stability.

The implication of this change is that if folks are uploading many files to a single location (ie: Folder), there would be a performance hit since each file needs to make a call to determine where it should be uploaded to.

cc: @linglp : This is also related to some discussion we were having while you were doing some upload testing. I know that we had seen some issues where this was hanging for some amount of time. This is just an fyi that I am planning to remove this.

Reply:
Thanks for letting me know, Bryan!

Reply:
@BryanFauble I think for uploading large files (~100GiB), it is best to use a sequential approach rather than a batch approach. When I tried uploading each file sequentially, the cache here is no longer a problem. I think you can keep it here for now.

Reply:
Sure. I can do that. I will keep you all posted about the result.